core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ActivityExtensions.kt

```diff
@@ -58,6 +65,45 @@ object ActivityExtensions {
     })
   }
 
+  fun Activity.openExternalUrl(
+    sharedPreferenceUtil: SharedPreferenceUtil,
+    alertDialogShower: AlertDialogShower,
+    intent: Intent
+  ) {
+    if (intent.resolveActivity(packageManager) != null) {
+      // Show popup with warning that this url is external and could lead to additional costs
+      // or may event not work when the user is offline.
+      if (urlIntentIsValid(intent) && sharedPreferenceUtil.prefExternalLinkPopup) {
+        requestOpenLink(alertDialogShower, intent, sharedPreferenceUtil)
+      } else {
+        openLink(intent)
+      }
+    } else {
+      val error = getString(R.string.no_reader_application_installed)
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+    }
+  }
+
+  private fun Activity.openLink(intent: Intent) {
+    ContextCompat.startActivity(this, intent, null)
```

Comment:
![image](https://user-images.githubusercontent.com/3358282/88770508-da2d4480-d175-11ea-9937-71c97a705ed1.png)
Pointless to use ContextCompat if we are just passing null

Reply:
Thanks, it's too easy to just copy previous code without thinking why it's different!